From eb1834d54f42f19ab44e4b67bc2ced1f99502f58 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 03 Nov 2021 15:08:08 +0200
Subject: [PATCH] [SW-0] gaudi2: remove set sram ext ld done

Instead of waiting for sram_init_done, wait for ext_ld_done to be
set was added to confirm that logic has finished copying the PHY FW
from PIF SRAM to the Synopsys PCS IP SRAM. This patch removes
pcie_phy_set_sram_ext_ld_done function which manually sets
sram_ext_ld_done.

Change-Id: If408c439851cdfd4db6d9cd2ab2740e5f95d5f3c
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index e5195b4..e65fc3d 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -367,11 +367,6 @@
 	} while (sram_ext_ld_done != 1);
 }
 
-static void pcie_phy_set_sram_ext_ld_done(void)
-{
-	regmap_writel(1, mmPCIE_PHY_PHY_SRAM_EXT_LD_DONE);
-}
-
 static void pcie_phy_set_mpll_a_b_ssc_enable(void)
 {
 
@@ -691,8 +686,6 @@
 
 	/* Wait for ext_ld_done before executing W/A */
 	pcie_wait_for_sram_ext_ld_done();
-
-	pcie_phy_set_sram_ext_ld_done();
 
 	/* Execute the Synopsys W/A */
 	regmap_writel(0xF, mmPCIE_PHY_CR_PARA_CTRL);
